Gifski/Preview/LockedCVPixelBuffer.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  copyCVPixelBuffers.swift
+//  Gifski
+//
+//  Created by Michael Mulet on 4/24/25.
+//
+
+import Foundation
+import CoreVideo
+
+/**
+ A CVPixelBuffer that has had it's base address locked via [CVPixelBufferLockBaseAddress](CVPixelBufferLockBaseAddress)
+ */
+protocol LockedCVPixelBuffer {
+	var buf: CVPixelBuffer {get}
+}
```

Comment:
I wonder if it would be simpler with just a helper method:

```swift
extension CVPixelBuffer {
 func withLocked<T>(flags: CVPixelBufferLockFlags = [], _ body: (UnsafeMutableRawPointer, Int, Int) throws -> T) rethrows -> T? {
  guard CVPixelBufferLockBaseAddress(self, flags) == kCVReturnSuccess,
        let base = CVPixelBufferGetBaseAddress(self) else { return nil }
  defer { CVPixelBufferUnlockBaseAddress(self, flags) }
  return try body(base,
                  CVPixelBufferGetBytesPerRow(self),
                  CVPixelBufferGetHeight(self))
 }
}
```